provider/shim/shim_test.go

```diff
@@ -1,29 +0,0 @@
-// Copyright 2023, Pulumi Corporation.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package shim
-
-import (
-	"context"
-	"testing"
-)
-
-// This checks that any runtime checks in the underlying provider (with patches) are passed.
```

Comment:
Can we bring this back in some form? This was a quick test that was introduced because it was useful.

Reply:
I brought this back as an `extraTest`. It now correctly ties into `sentinel` and `publish`, preventing merging and publishing, respectively, when the test fails.